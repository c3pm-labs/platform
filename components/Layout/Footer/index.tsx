import { makeStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';
import Link from 'next/link';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'space-around',
    width: '100%',
    minHeight: '6em',
    backgroundColor: theme.palette.grey[100],
    borderTop: `1px solid ${theme.palette.grey[200]}`,
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    paddingLeft: theme.spacing(2),
  },
  textContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    margin: theme.spacing(2),
    '& span': {
      fontSize: 14,
      fontWeight: 500,
      marginBottom: theme.spacing(1),
    },
  },
  text: {
    textDecoration: 'none',
    fontSize: 14,
    color: theme.palette.grey[500],
    marginBottom: 3,
  },
}));

function Footer(): JSX.Element {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <div className={classes.textContainer}>
        <span>General Information</span>
        <a
          href="https://docs.c3pm.io"
          className={classes.text}
        >
          <Typography className={classes.text}>Docs</Typography>
        </a>
        <a
          href="https://github.com/gabrielcolson/c3pm/"
          className={classes.text}
        >
          <Typography className={classes.text}>Github</Typography>
        </a>
        <a
          href="https://github.com/gabrielcolson/c3pm/releases"
          className={classes.text}
        >
          <Typography className={classes.text}>Releases</Typography>
        </a>
      </div>
      <div className={classes.textContainer}>
        <span>Contact us</span>
        <a
          href="mailto:contact@c3pm.io"
          className={classes.text}
        >
          <Typography className={classes.text}>
            contact@c3pm.io
          </Typography>
        </a>
        <a
          href="https://github.com/gabrielcolson/c3pm/issues/new/choose"
          className={classes.text}
        >
          <Typography className={classes.text}>Issues</Typography>
        </a>
        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
        <Link href="/team"><a className={classes.text}>Team</a></Link>
      </div>
    </div>
  );
}

export default Footer;
