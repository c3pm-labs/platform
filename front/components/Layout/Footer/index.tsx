import { makeStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';
import Link from 'next/link';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    alignItems: 'flex-start',
    justifyContent: 'space-around',
    width: '100%',
    minHeight: '6em',
    backgroundColor: theme.palette.grey[100],
    borderTop: `1px solid ${theme.palette.grey[200]}`,
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    paddingLeft: theme.spacing(2),
  },
  textContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
    margin: theme.spacing(2),
    '& span': {
      fontSize: 14,
      fontWeight: 500,
      marginBottom: theme.spacing(1),
    },
  },
  text: {
    textDecoration: 'none',
    fontSize: 14,
    color: theme.palette.grey[500],
    marginBottom: 3,
  },
}));

function Footer(): JSX.Element {
  const classes = useStyles();
  const linkElement = (link: string, name: string) => (
    <a
      target="_blank"
      rel="noopener noreferrer"
      className={classes.text}
      href={link}
    >
      <Typography className={classes.text}>{name}</Typography>
    </a>
  );

  return (
    <div className={classes.container}>
      <div className={classes.textContainer}>
        <span>General Information</span>
        {linkElement('https://docs.c3pm.io', 'Docs')}
        {linkElement('https://github.com/c3pm-labs/', 'Github')}
        {linkElement('https://github.com/c3pm-labs/c3pm/releases', 'Releases')}
      </div>
      <div className={classes.textContainer}>
        <span>Contact us</span>
        {linkElement('mailto:contact@c3pm.io', 'contact@c3pm.io')}
        {linkElement('https://github.com/c3pm-labs/c3pm/issues/new/choose', 'Submit an issue')}
      </div>
    </div>
  );
}

export default Footer;
