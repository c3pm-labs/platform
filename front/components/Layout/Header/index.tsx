import { Hidden } from '@material-ui/core';
import useScrollTrigger from '@material-ui/core/useScrollTrigger';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';

import RightNav from 'components/Layout/Header/RightNav';
import Logo from 'components/Logo';

import SearchBar from './SearchBar';

const useStyles = makeStyles((theme) => ({
  appBar: {
    backgroundColor: theme.palette.background.default,
  },
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
  },
  search: {
    maxWidth: 480,
    flex: 1,
    margin: `0 ${theme.spacing(3)}px`,
    [theme.breakpoints.down('xs')]: {
      margin: `0 ${theme.spacing(2)}px`,
    },
  },
}));

function Header(): JSX.Element {
  const classes = useStyles();
  const elevationTrigger = useScrollTrigger({ threshold: 10, disableHysteresis: true });

  return (
    <AppBar
      elevation={elevationTrigger ? 3 : 0}
      className={classes.appBar}
      position="sticky"
      color="default"
    >
      <Toolbar className={classes.container}>
        <Hidden implementation="css" smUp>
          <Logo type="mini" />
        </Hidden>
        <Hidden implementation="css" xsDown>
          <Logo type="classic" size="sm" />
        </Hidden>
        <SearchBar
          dataTestId="search-header"
          className={classes.search}
        />
        <RightNav />
      </Toolbar>
    </AppBar>
  );
}

export default Header;
