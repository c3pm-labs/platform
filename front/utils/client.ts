import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'

const httpLink = createHttpLink({
  uri: process.env.GRAPHQL_URL,
  credentials: 'include',
})

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
    },
  }
})

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
  ssrMode: typeof window === 'undefined',
})

export default client
