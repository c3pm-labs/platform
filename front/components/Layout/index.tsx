import { ReactNode } from 'react';
import { makeStyles } from '@material-ui/core/styles';

import Header from './Header';
import Footer from './Footer';

const useStyles = makeStyles(() => ({
  container: {
    minHeight: '100vh',
    backgroundImage: 'linear-gradient(124.57deg, rgba(38, 179, 239, 0.2) 37.9%, rgba(255, 112, 68, 0.2) 74.33%)',
  },
}));

export interface LayoutProps {
  children: ReactNode;
}

function Layout({ children }: LayoutProps): JSX.Element {
  const classes = useStyles();

  return (
    <>
      <div className={classes.container}>
        <Header />
        {children}
      </div>
      <Footer />
    </>
  );
}

export default Layout;
