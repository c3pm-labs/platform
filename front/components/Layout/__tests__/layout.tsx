import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import React from 'react';

import Layout from '..';

// eslint-disable-next-line
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

test('Layout', () => {
  const mocks = [];

  useRouter.mockImplementation(() => ({
    route: '',
    pathname: '',
    query: '',
    asPath: '',
  }));

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Layout>Toto</Layout>
    </MockedProvider>,
  );

  expect(container).toMatchSnapshot();
});
