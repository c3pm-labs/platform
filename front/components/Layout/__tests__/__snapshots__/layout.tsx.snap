// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorDefault makeStyles-appBar-2 MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-container-3 MuiToolbar-gutters"
      >
        <div
          class="PrivateHiddenCss-smUp-9"
        >
          <a
            class="makeStyles-link-20"
            href="/"
          >
            <div
              class="makeStyles-logo-22"
            >
              <img
                alt="mini-sm c3pm logo"
                height="40"
                src="/assets/c3pm_3.png"
                width="auto"
              />
            </div>
          </a>
        </div>
        <div
          class="PrivateHiddenCss-xsDown-7"
        >
          <a
            class="makeStyles-link-20"
            href="/"
          >
            <div
              class="makeStyles-logo-22"
            >
              <img
                alt="classic-sm c3pm logo"
                height="auto"
                src="/assets/c3pm.png"
                width="100"
              />
            </div>
          </a>
        </div>
        <form
          action="#"
          class="makeStyles-search-4"
          data-testid="SearchBarInput"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="search"
                name="search"
                placeholder="search..."
                type="search"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-26 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-27"
                  style="width: 0.01px;"
                >
                  <span>
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </form>
        <div>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-default-32 makeStyles-primary-33 makeStyles-primary-37 makeStyles-marginRight-31 MuiButton-disableElevation"
            href="/login"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Sign in
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-32 makeStyles-primary-33 makeStyles-primary-41 MuiButton-disableElevation"
            href="/register"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Sign up
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </header>
    Toto
  </div>
  <div
    class="makeStyles-container-45"
  >
    <div
      class="makeStyles-textContainer-46"
    >
      <span>
        General Information
      </span>
      <a
        class="makeStyles-text-47"
        href="https://docs.c3pm.io"
      >
        <p
          class="MuiTypography-root makeStyles-text-47 MuiTypography-body1"
        >
          Docs
        </p>
      </a>
      <a
        class="makeStyles-text-47"
        href="https://github.com/c3pm-labs/"
      >
        <p
          class="MuiTypography-root makeStyles-text-47 MuiTypography-body1"
        >
          Github
        </p>
      </a>
      <a
        class="makeStyles-text-47"
        href="https://github.com/c3pm-labs/c3pm/releases"
      >
        <p
          class="MuiTypography-root makeStyles-text-47 MuiTypography-body1"
        >
          Releases
        </p>
      </a>
    </div>
    <div
      class="makeStyles-textContainer-46"
    >
      <span>
        Contact us
      </span>
      <a
        class="makeStyles-text-47"
        href="mailto:contact@c3pm.io"
      >
        <p
          class="MuiTypography-root makeStyles-text-47 MuiTypography-body1"
        >
          contact@c3pm.io
        </p>
      </a>
      <a
        class="makeStyles-text-47"
        href="https://github.com/c3pm-labs/c3pm/issues/new"
      >
        <p
          class="MuiTypography-root makeStyles-text-47 MuiTypography-body1"
        >
          Issues
        </p>
      </a>
      <a
        class="makeStyles-text-47"
        href="/team"
      >
        Team
      </a>
    </div>
  </div>
</div>
`;
